tests/test_odl.py

```diff
@@ -1685,59 +1879,319 @@ def do_get(url, headers):
+        assert imported_license.is_inactive is True
+
+    def test_odl_importer_reimport_expired_licenses(self, import_templated):
+        license_expiry = dateutil.parser.parse("2021-01-01T00:01:00+00:00")
+        licenses = [
+            LicenseInfoHelper(
+                license=LicenseHelper(concurrency=1, expires=license_expiry),
+                available=1,
+            )
+        ]
+
+        # First import the license when it is not expired
+        with freeze_time(license_expiry - datetime.timedelta(days=1)):
+
+            # Import the test feed.
+            (
+                imported_editions,
+                imported_pools,
+                imported_works,
+                failures,
+            ) = import_templated(licenses)
+
+            # The importer created 1 edition and 1 work with no failures.
+            assert failures == {}
+            assert len(imported_editions) == 1
+            assert len(imported_works) == 1
+            assert len(imported_pools) == 1
+
+            # Ensure that the license pool was successfully created, with available copies.
+            [imported_pool] = imported_pools
+            assert imported_pool.licenses_owned == 1
+            assert imported_pool.licenses_available == 1
+            assert len(imported_pool.licenses) == 1
+
+            # Ensure the license was imported and is not expired.
+            [imported_license] = imported_pool.licenses
+            assert imported_license.is_inactive is False
+
+        # Reimport the license when it is expired
+        with freeze_time(license_expiry + datetime.timedelta(days=1)):
+
+            # Import the test feed.
+            (
+                imported_editions,
+                imported_pools,
+                imported_works,
+                failures,
+            ) = import_templated(licenses)
+
+            # The importer created 1 edition and 1 work with no failures.
+            assert failures == {}
+            assert len(imported_editions) == 1
+            assert len(imported_works) == 1
+            assert len(imported_pools) == 1
+
+            # Ensure that the license pool was successfully created, with no available copies.
+            [imported_pool] = imported_pools
+            assert imported_pool.licenses_owned == 0
+            assert imported_pool.licenses_available == 0
+            assert len(imported_pool.licenses) == 1
+
+            # Ensure the license was imported and is expired.
+            [imported_license] = imported_pool.licenses
+            assert imported_license.is_inactive is True
+
+    @freeze_time("2021-01-01T00:00:00+00:00")
+    def test_odl_importer_multiple_expired_licenses(self, import_templated):
+        """Ensure ODLImporter imports expired licenses
+        and does not count them in the total number of available licenses."""
+
+        # 1.1. Import the test feed with three expired ODL licenses and two valid licenses.
+        invalid = [
```

Comment:
This was nagging me, too, since none of these licenses are really invalid. I would suggest calling these two values `active_licenses` and `inactive_licenses`, instead of `valid` and `invalid`.

Reply:
Agreed. Renamed now.